Week3/week3_homework/card_validator.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const test_valid_input = '6698666658666166';
+const test_invalid_input = 'a9224533219c0112';
+
+// #- Number must be 16 digits
+const digits_count = 16;
+
+function validate_number_of_digits(input) {
+  return input.length === digits_count ? true : false;
+}
+
+// #- all of them must be numbers
+const is_number_validation = /^[0-9]+$/;
+
+function validate_is_number(input) {
+  return is_number_validation.test(input) ? true : false;
+}
+
+//#- You must have at least two different digits represented (all of the digits cannot be the same)
+function validate_frequency(input) {
+  let counter = 0;
+  const firstChar = input[0];
+
+  for (const x of input) {
+    if (firstChar === x) counter++;
+  }
+
+  return counter >= digits_count ? false : true;
```

Comment:
Ah. I would like to introduce a new feature in JavaScript called **Sets** which will help you in significantly reducing the amount of code for this logic - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Please have a look 🙂

Reply:
I'll have a look, 
thanks